src/Features/LanguageServer/ProtocolUnitTests/Completion/CompletionResolveTests.cs

```diff
@@ -31,20 +31,53 @@ void M()
-            var completionParams = CreateCompletionParams(locations["caret"].Single(), LSP.VSCompletionInvokeKind.Explicit, "\0", LSP.CompletionTriggerKind.Invoked);
+            var completionParams = CreateCompletionParams(
+                locations["caret"].Single(), LSP.VSCompletionInvokeKind.Explicit, "\0", LSP.CompletionTriggerKind.Invoked);
             var document = testLspServer.GetCurrentSolution().Projects.First().Documents.First();
 
             var completionItem = await CreateCompletionItemAsync(
-                "A", LSP.CompletionItemKind.Class, tags, completionParams, document, commitCharacters: CompletionRules.Default.DefaultCommitCharacters).ConfigureAwait(false);
+                label: "A", LSP.CompletionItemKind.Class, tags, completionParams, document,
+                commitCharacters: CompletionRules.Default.DefaultCommitCharacters, insertText: "A").ConfigureAwait(false);
             var description = new ClassifiedTextElement(CreateClassifiedTextRunForClass("A"));
             var clientCapabilities = new LSP.VSClientCapabilities { SupportsVisualStudioExtensions = true };
 
             var expected = CreateResolvedCompletionItem(completionItem, description, "class A", null);
 
-            var results = (LSP.VSCompletionItem)await RunResolveCompletionItemAsync(testLspServer, completionItem, clientCapabilities).ConfigureAwait(false);
+            var results = (LSP.VSCompletionItem)await RunResolveCompletionItemAsync(
+                testLspServer, completionItem, clientCapabilities).ConfigureAwait(false);
             AssertJsonEquals(expected, results);
         }
 
+        [Fact]
+        public async Task TestResolveOverridesCompletionItemAsync()
+        {
+            var markup =
+@"abstract class A
+{
+    public abstract void M();
+}
+
+class B : A
+{
+    override {|caret:|}
+}";
+            using var testLspServer = CreateTestLspServer(markup, out var locations);
+            var tags = new string[] { "Method", "Public" };
+            var completionParams = CreateCompletionParams(
+                locations["caret"].Single(), LSP.VSCompletionInvokeKind.Explicit, "\0", LSP.CompletionTriggerKind.Invoked);
+            var document = testLspServer.GetCurrentSolution().Projects.First().Documents.First();
+
+            var completionItem = await CreateCompletionItemAsync(
+                label: "M()", LSP.CompletionItemKind.Method, tags, completionParams, document,
+                commitCharacters: CompletionRules.Default.DefaultCommitCharacters).ConfigureAwait(false);
+            var clientCapabilities = new LSP.VSClientCapabilities { SupportsVisualStudioExtensions = true };
+
+            var results = (LSP.VSCompletionItem)await RunResolveCompletionItemAsync(
+                testLspServer, completionItem, clientCapabilities).ConfigureAwait(false);
+            Assert.NotNull(results.TextEdit);
+            Assert.Null(results.InsertText);
+        }
```

Comment:
i'd like an actual verification that afteer we apply the textedit we get teh right result (including caret position).